zynqmp.mk

```diff
@@ -1,88 +1,171 @@
-PLATFORM		?= zcu102
-BSP_PATH		?= ../xilinx-${PLATFORM}-v$(PETALINUX_VER)-final.bsp
-PRJ_PATH		?= ../petalinux-optee-$(PLATFORM)
-OPTEE_VER		?= latest
-
-PYTHON_PATH	?= ${PRJ_PATH}/project-spec/meta-user/recipes-devtools/python
-
-define set_cfg
-	@sed -i 's/$(1)=.*/$(1)=$(2)/' $(3)
-endef
-
-define set_optee_version
-	@if [ "$(1)" != "latest" ]; then \
-		echo 'OPTEE_VERSION ?= "$(1)"' > $(2); \
-		echo 'SRCREV ?= "$(1)"' >> $(2); \
-	else \
-		echo 'OPTEE_VERSION ?= "latest"' > $(2); \
-		echo 'SRCREV ?= "$${AUTOREV}"' >> $(2); \
-	fi
-endef
-
-ifeq ($(PLATFORM),ultra96-reva)
-	ZYNQMP_CONSOLE=cadence1
+################################################################################
+# Following variables defines how the NS_USER (Non Secure User - Client
+# Application), NS_KERNEL (Non Secure Kernel), S_KERNEL (Secure Kernel) and
+# S_USER (Secure User - TA) are compiled
+################################################################################
+override COMPILE_NS_USER   := 64
+override COMPILE_NS_KERNEL := 64
+override COMPILE_S_USER    := 64
+override COMPILE_S_KERNEL  := 64
+
+PLATFORM = zynqmp-zcu102
+OPTEE_OS_PLATFORM = zynqmp-zcu102
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+TF_A_PATH		?= $(ROOT)/arm-trusted-firmware
+U-BOOT_PATH		?= $(ROOT)/u-boot-xlnx
+BOOTGEN_PATH		?= $(ROOT)/bootgen
+LINUX_PATH		?= $(ROOT)/linux-xlnx
+
+include common.mk
+
+################################################################################
+# Targets
+################################################################################
+
+all: bootimage fitimage
+clean: bootimage-clean fitimage-clean
+
+include toolchain.mk
+
+###############################################################################
+# Boot Image
+###############################################################################
+FIRMWARE_TARBALL = $(subst zynqmp-,2021.1-,$(PLATFORM))-release.tar.xz
+
+ifeq ("$(wildcard ../$(FIRMWARE_TARBALL))","")
+$(warning Release image tarball no present ../$(FIRMWARE_TARBALL))
```

Comment:
Is it still possible to complete the build without this?
I assume this is the binary which needs to be downloaded from the wiki page. Please add a link in the error message or at least here in the makefile.

Reply:
Or even add a target to download the file?

Reply:
Hi @jenswi-linaro and @jforissier, I added the link in the documentation repo update (did not pull request yet) but I think adding the URL here would also take sense. The main issue is that the wiki just provides a link to a Xilinx portal that requires account access, therefore no way to download as part of Makefile. I understand that might not be ideal but it is best than downloading few GB tools :)

The positive side is that the build can be performed even without this file, the only difference is that the bootable image will not be created but every single artifact will be built (OPTEE, Kernel, Rootfs.....)

Reply:
A problem with not erroring out here is that the log entry will be lost in a jungle of log entries.
Compare with the FVP setup where we error out if the model isn't available, even though everything could have been built.

Reply:
I see, actually my initial approach was that one. However I still see the value on building the sources specially for regression purposes. I mean, I just wanted to allow building OP-TEE without requiring to download binary blobs from Xilinx portal, mostly thinking on those not using Xilinx devices but doing some regression testing.

Maybe I can move the warning message print to the "all" target in a way that is printed just at the end, in a way that is more visible why the BOOT.bin file is missing from the deployment folder.

If you still think that is just better to stop the build if the firmware is missing, I can do that as well.

Reply:
Perhaps we should consider BOOT.bin is a dependency of the next step (`make flash` or whatever), and just ignore it completely until that step.